### PR TITLE
Refactor MC/Failsafe/add Subsystems

### PR DESCRIPTION
**Pull request by neersighted**
:cl:
tweak: MC tab reworked, more items made available.
add: Admins can now click entries in the MC tab to debug.
/:cl:

Demo

### CODE REVIEW

File: code/controllers/admin.dm [hunk 1]
```diff
@@ -1,54 +1,73 @@
...
+/obj/effect/statclick/New(text, target)
+	src.name = text
+	src.target = target
+
+/obj/effect/statclick/proc/update(text)
+	src.name = text
+	return src
+
+/obj/effect/statclick
+	var/class
```
Comment (YotaXP) [inline code/controllers/admin.dm]: Why are the `target` and `class` vars not defined in the same block?

File: code/controllers/admin.dm [hunk 2]
```diff
@@ -1,54 +1,73 @@
...
-	message_admins("Admin [key_name_admin(usr)] is debugging the [controller] controller.")
-	return
+//TODO: rewrite and standardise all controller datums to the datum/controller type
+//TODO: allow all controllers to be deleted for clean restarts (see WIP master controller stuff) - MC done - lighting done
+
+/obj/effect/statclick
+	var/target
+
+/obj/effect/statclick/New(text, target)
+	src.name = text
```
Comment (RemieRichards) [inline code/controllers/admin.dm]: dont use ```src``` like this

**Comment by ike709** [comment]
1. Real DEFCON goes from 5 (least severe) to 1 (most severe) but I mean this is probably simpler to code so I don't really care.

2. It bothers me that the subsystems aren't listed alphabetically, even if I'll never see the MC tab

**Comment by neersighted** [comment]
Yeah, I wanted to make defcon a bit better... I want it to mirror the real defcon...
As far as the alphabetical sort, I tried to make it work... still working on that...

**Comment by duncathan** [comment]
Should be listed in order of importance imo

File: code/controllers/subsystem/assets.dm [hunk 3]
```diff
@@ -0,0 +1,23 @@
+var/datum/subsystem/assets/SSasset
+
+/datum/subsystem/assets
+	name = "Assets"
+	priority = 19
```
Comment (MrStonedOne) [inline code/controllers/subsystem/assets.dm]: You'll want to set this to a lower priority... other wise assets made by lower priority subsystems won't get sent to world start clients.

in fact, the default would be best.

File: code/controllers/failsafe.dm [hunk 4]
```diff
@@ -1,43 +1,69 @@
...
 /datum/controller/failsafe // This thing pretty much just keeps poking the master controller
-	var/processing_interval = 100	//poke the MC every 10 seconds - set to 0 to disable
+	name = "Failsafe"
+
+	// The length of time to check on the MC (in deciseconds).
+	// Set to 0 to disable.
+	var/processing_interval = 100
+	// The alert level. For every failed poke, we drop a DEFCON level. Once we hit DEFCON 1, restart the MC.
+	var/defcon = 0
+	var/list/defcon_pretty = list(5, 4, 3, 2, 1)
```
Comment (GinjaNinja32) [inline code/controllers/failsafe.dm]: you don't appear to actually use the list anywhere

Comment (neersighted) [inline code/controllers/failsafe.dm]: Oop, that was from a failed fix, thanks.

**Pull request merged**